### PR TITLE
feat: add deprecation warnings for legacy commands

### PR DESCRIPTION
**Pull request by ysndr**

### CODE REVIEW

**Comment by ysndr** [comment]
Not sure if I catched all deprecated commands, please call me out if i missed some @flox/cli

File: crates/flox/src/commands/mod.rs [hunk 1]
```diff
@@ -435,13 +437,49 @@ impl AdditionalCommands {
...
         Ok(())
     }
 }
 
+fn check_deprecated_commands(commands: &Commands) {
+    match commands {
+        Commands::Development(_) => { /* none deprecated */ },
+        Commands::Sharing(_) => { /* none deprecated */ },
+        Commands::Additional(additional_commands) => match additional_commands {
+            AdditionalCommands::Build(_) => deprecate_command("build"),
```
Comment (ghudgins) [inline crates/flox/src/commands/mod.rs:458]: if build is deprecated shouldn't publish be as well

Comment (ysndr) [reply crates/flox/src/commands/mod.rs:463]: added publish and init-package

File: crates/flox/src/commands/mod.rs [hunk 2]
```diff
@@ -435,13 +437,56 @@ impl AdditionalCommands {
...
+            InternalCommands::Publish(_) => deprecate_command("publish"),
+            InternalCommands::InitPackage(_) => deprecate_command("init-package"),
+            _ => { /* not deprecated */ },
+        },
+    }
+}
+
+fn deprecate_command(cmd: &str) {
+    warn!("----------------------------------------- deprecated command -----------------------------------------");
+    warn!("`flox {cmd}` is being depricated and will be removed in the next release");
```
Comment (mkenigs) [inline crates/flox/src/commands/mod.rs:485]:
```suggestion
    warn!("`flox {cmd}` is being deprecated and will be removed in the next release");
```

File: crates/flox/src/commands/mod.rs [hunk 3]
```diff
@@ -435,13 +437,56 @@ impl AdditionalCommands {
...
-            AdditionalCommands::PrintDevEnv(args) => args.handle(config, flox).await?,
-            AdditionalCommands::Shell(args) => args.handle(config, flox).await?,
+            AdditionalCommands::PrintDevEnv(args) => {
+                deprecate_command("print-dev-env");
+                args.handle(config, flox).await?
+            },
+            AdditionalCommands::Shell(args) => {
+                deprecate_command("shell");
+                args.handle(config, flox).await?
+            },
```
Comment (mkenigs) [inline crates/flox/src/commands/mod.rs:447]: This is printing the deprecation warning a second time

**Review by mkenigs — CHANGES_REQUESTED** [review]
See comments

File: crates/flox/src/commands/mod.rs (hunk 3 above)
Comment (ysndr) [reply crates/flox/src/commands/mod.rs:447]: oh yeah forgot to remove those -.-

Comment (ysndr) [reply crates/flox/src/commands/mod.rs]: done in https://github.com/flox/flox/commit/60d9f11ae7ba41b8a39b0aadb80c4c7947222182

**Review by mkenigs — APPROVED** [review]

**Comment by mkenigs** [comment]
User facing change: adds deprecation warnings for a number of commands

**Review by ghudgins — APPROVED** [review]

**Comment by mkenigs** [comment]
Test output needs to be updated (or maybe just drop the tests)

**Comment by mkenigs** [comment]
> User facing change: adds deprecation warnings for a number of commands

Specifically build, subscribe, unsubscribe, channels, print-dev-env, shell, bundle, flake, eval, develop, publish, init-package

**Review by bryanhonof — APPROVED** [review]

**Comment by ysndr** [comment]
> Test output needs to be updated (or maybe just drop the tests)

Its only the eval test that's failing on account of the message

**Comment by mkenigs** [comment]
Rerunning macOS tests as they flaked on flushing

**Pull request merged**